src/common/config/can_show_page.ts

```diff
@@ -0,0 +1,18 @@
+import { isComponentLoaded } from "./is_component_loaded";
+import { PageNavigation } from "../../layouts/hass-tabs-subpage";
+import { HomeAssistant } from "../../types";
+
+export const canShowPage = (hass: HomeAssistant, page: PageNavigation) => {
+  return (
+    (isCore(page) || isLoadedIntegration(hass, page)) &&
+    !hideAdvancedPage(hass, page)
+  );
+};
+
+const isLoadedIntegration = (hass: HomeAssistant, page: PageNavigation) =>
+  !page.component || isComponentLoaded(hass, page.component);
+const isCore = (page: PageNavigation) => page.core;
+const isAdvancedPage = (page: PageNavigation) => page.advancedOnly;
+const userWantsAdvanced = (hass: HomeAssistant) => hass.userData?.showAdvanced;
+const hideAdvancedPage = (hass: HomeAssistant, page: PageNavigation) =>
+  isAdvancedPage(page) && !userWantsAdvanced(hass);
```

Comment:
Maybe over-kill, but I personally preferred well-named one-liner functions if it helps make the logic clearer.